src/InfluxDB/Driver/TCP.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * @author Stephen "TheCodeAssassin" Hoogendijk
+ */
+
+namespace InfluxDB\Driver;
+
+/**
+ * Class UDP
+ *
+ * @package InfluxDB\Driver
+ */
+class TCP extends AbstractSocketDriver implements DriverInterface
+{
+
+    private $result;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isSuccess()
+    {
+        return (bool) $this->result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createStream()
+    {
+        $host = sprintf('tcp://%s:%d', $this->config['host'], $this->config['port']);
+
+        // stream the data using TCP and suppress any errors
+        $this->stream = @stream_socket_client($host);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doWrite($data)
+    {
+        $this->result = false !== @fwrite($this->stream, "$data\n");
```

Comment:
An issue can occur when multiple messages are written into TCP socket:
```
telegraf_1_279d9d02c69e | 2019-02-27T10:54:58Z E! [inputs.socket_listener]: Error in plugin: unable to parse incoming line: metric parse error: expected field at offset 49: "xxx,tag=tag_value parsed=0i,processing_time=85.6xxx,tag=tag_value parsed=0i...
```
Each message has to end with newline.